### PR TITLE
{REST} Capture SSLError in handle_exception

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  

Split from #14588.

In `handle_exception`, capture `requests.exceptions.SSLError` which is generated by directly calling `requests`'s methods, and provide a better error message.

**Testing Guide**  

Enable Fiddler HTTPS decryption and make `az rest` call like:

```
az rest -u 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups/fy?api-version=2018-02-01'
```

Output:
> Request failed: Certificate verification failed. This typically happens when using Azure CLI behind a proxy that intercepts traffic with a self-signed certificate. Please add this certificate to the trusted CA bundle. More info: https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively#work-behind-a-proxy.
> 
> Error detail: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups/fy?api-version=2018-02-01 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])")))

### CODE REVIEW

**Comment by yonzhan** [comment]
REST

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -802,19 +804,17 @@ def send_raw_request(cli_ctx, method, url, headers=None, uri_parameters=None,  #
...
-        req = Request(method=method, url=url, headers=headers, params=uri_parameters, data=body)
-        prepped = s.prepare_request(req)
-
-        # Merge environment settings into session
-        settings = s.merge_environment_settings(prepped.url, {}, None, not should_disable_connection_verify(), None)
-        _log_request(prepped)
-        r = s.send(prepped, **settings)
-        _log_response(r)
-    except Exception as ex:  # pylint: disable=broad-except
-        raise CLIError(ex)
```
Comment (zhoxing-ms) [inline src/azure-cli-core/azure/cli/core/util.py:817]: May I ask why the CLIError is not wrapped here any more?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/util.py:817]: If we wrap the exception with `CLIError`, the exception type will be lost and 
https://github.com/Azure/azure-cli/blob/a0f8c8ecbec012b07c00441639f5866043b3c218/src/azure-cli-core/azure/cli/core/util.py#L82
won't be hit anymore.

We are leaving the error handing to `handle_exception` instead.

**Review by houk-ms — APPROVED** [review]
Looks good. Add reference #14855

File: src/azure-cli-core/azure/cli/core/util.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli-core/azure/cli/core/util.py:817]: Got it, thx~

**Review by zhoxing-ms — APPROVED** [review]
LGTM

File: src/azure-cli-core/azure/cli/core/util.py [hunk 2]
```diff
@@ -807,19 +809,17 @@ def send_raw_request(cli_ctx, method, url, headers=None, uri_parameters=None,  #
...
-        s = Session()
-        req = Request(method=method, url=url, headers=headers, params=uri_parameters, data=body)
-        prepped = s.prepare_request(req)
-
-        # Merge environment settings into session
-        settings = s.merge_environment_settings(prepped.url, {}, None, not should_disable_connection_verify(), None)
-        _log_request(prepped)
-        r = s.send(prepped, **settings)
-        _log_response(r)
-    except Exception as ex:  # pylint: disable=broad-except
```
Comment (arrownj) [inline src/azure-cli-core/azure/cli/core/util.py:821]: Is it always `SSLError` ?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/util.py:821]: No. We throw any exceptions and let `handle_exception` handle it.

**Review by arrownj — APPROVED** [review]

**Pull request merged**